### PR TITLE
feat(Breadcrumb): new component

### LINKED ISSUE #184: [Feat]:New Breadcrumb component
**Description**: The `Breadcrumb `component provides a navigation aid that helps users understand their location within the hierarchy of a website or application. It displays a trail of links indicating the current page's path, allowing users to easily navigate back to previous levels.

**Requirements**:

1. **Component Implementation:**

- Implement the `Breadcrumb `component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and separators.
- Support dynamic generation of breadcrumbs from a route path and provide custom separators (e.g., /, >, |).

2. **Unit Tests:**

- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of breadcrumb links and separators.
- Test for edge cases such as missing links or unsupported separator types.

3. **Storybook Stories:**

- Create a Storybook story for the Breadcrumb component.
- Showcase different breadcrumb structures (e.g., with multiple levels, with custom separators).
- Create controls for the props so users can interact with the Breadcrumb component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=141-9&node-type=canvas&t=BW68yyejj2aaHpzs-0)

**Checklist**:

- [ ] Component is implemented in TypeScript.
- [ ] Styling follows SCSS using BEM methodology.
- [ ] Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and separators.
- [ ] Unit tests written and cover different states and edge cases.
- [ ] Storybook stories created with examples of all props and breadcrumb types.
- [ ] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by GaroGabrielyan**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1295 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-whutrlaxrc.chromatic.com/

### CODE REVIEW

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    useLayoutEffect(() => {
+        const config = calculateVisibilityConfig(containerRef.current, measureListRef.current, itemsCount);
+        setVisibilityConfig(config);
+    }, [containerWidth, itemsCount, items]);
+
+    const { fitAll, firstCount, lastCount } = visibilityConfig;
+    const shouldShowEllipsis = itemsCount > 1 && !fitAll;
+
+    const { visibleFirstItems, visibleLastItems, menuItems } = useMemo(() => {
+        if (!items || itemsCount === 0) {
```
Comment (alberthovhannisyan1) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx]: !items is additional

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+            return { visibleFirstItems: [], visibleLastItems: [], menuItems: [] };
+        }
+
+        const total = itemsCount;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx]: Remove this variable and use just itemsCount

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+    const showEllipsisTrigger = shouldShowEllipsis && menuItems.length > 0;
+
+    return (
+        <div ref={containerRef} className={classNames("breadcrumb", className)}>
+            {/* Hidden measurement DOM: all items + ellipsis for width calculation */}
+            <div
+                className="breadcrumb__measure"
+                aria-hidden
+                style={{
```
Comment (alberthovhannisyan1) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx]: Why we need inline styles here ?

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+            <BreadcrumbItem {...props} />
+        </BreadcrumbContext.Provider>
+    );
+};
+
+/**
+ * Breadcrumb component is a navigational aid that displays the user's current location within a website or application. It provides a trail of links back to the starting or entry point, allowing users to easily navigate through the hierarchical structure of the site. Breadcrumbs enhance usability by offering a clear path for users to trace their steps and return to previous sections.
+ */
+const Breadcrumb: FC<IBreadcrumbProps> = ({ className, items = [], iconOnly = false, render, onClick }) => {
+    const [menuPropsForPopover, setMenuPropsForPopover] = useState<Record<string, unknown>>({});
```
Comment (alberthovhannisyan1) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx]: IG type is not necessary here

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            visibleLastItems: visibleLast,
+            menuItems: menuItemsSlice
+        };
+    }, [items, itemsCount, shouldShowEllipsis, firstCount, lastCount]);
+
+    const menuSelectHandler = (menuItem: IMenuItemProps) => {
+        const selectedItem =
+            typeof menuItem.id === "number"
+                ? menuItems[menuItem.id]
+                : menuItems.find((item) => (item.path || item.title) === menuItem.id);
```
Comment (alberthovhannisyan1) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx]: Can we destructurize the path and title from item directly ?

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+                    position: "absolute",
+                    visibility: "hidden",
+                    pointerEvents: "none",
+                    left: 0,
+                    top: 0
+                }}
+            >
+                <ul ref={measureListRef} className="breadcrumb__list">
+                    {items.map((item, index) => {
+                        const isLast = index === items.length - 1;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx:206]: Move this logic upper, instead of creating new variables add new [key]: values directly into each item object.
Here left just mapping and render of the component

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx (hunk 2 above)
Comment (GaroGabrielyan) [reply src/components/molecules/Breadcrumb/Breadcrumb.tsx]: we can just rename fore more meaningful name

File: src/components/molecules/Breadcrumb/Breadcrumb.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    title: "Molecules/Breadcrumb",
+    component: Breadcrumb,
+    argTypes: {
+        className: args({ control: "false", ...propCategory.appearance }),
+        items: args({ control: "false", ...propCategory.content }),
+        iconOnly: args({ control: "boolean", ...propCategory.appearance }),
+        render: args({ control: "false", ...propCategory.content }),
+        onClick: args({ control: "false", ...propCategory.action })
+    },
+    args: {
```
Comment (HaykKirakosyanSoft) [inline src/components/molecules/Breadcrumb/Breadcrumb.stories.tsx]: please remove args empty object

File: src/components/molecules/Breadcrumb/Breadcrumb.tsx [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+                        </Tooltip>
+                        <Menu
+                            onChange={menuSelectHandler}
+                            setPropsForPopover={setMenuPropsForPopover}
+                            position="bottom-right"
+                        >
+                            {menuItems.map((item, index) => {
+                                const itemId = (item.path || item.title) as string;
+                                const key = `${itemId}-${index}`;
+                                const linkData = {
```
Comment (HaykKirakosyanSoft) [inline src/components/molecules/Breadcrumb/Breadcrumb.tsx]: we are creating a brand-new object in memory on every single re-rende. If we map over menuItems inside the useMemo to pre-attach this linkData, we can just pass it directly here without recalculating it

File: src/components/molecules/Breadcrumb/BreadcrumbItem.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    );
+
+    const linkData = { path, title, isActive: isLastItem, Icon };
+
+    const interactiveElement = (() => {
+        if (render && path && !isLastItem) {
+            const renderedElement = render(linkData);
+            if (isValidElement(renderedElement)) {
+                const propsToApply = {
+                    className: classNames("breadcrumb__link", {
```
Comment (HaykKirakosyanSoft) [inline src/components/molecules/Breadcrumb/BreadcrumbItem.tsx:68]: we use cloneElement here to apply our breadcrumb classes and onClick handler, we are completely overwriting any className or onClick the user might pass his own classes to their custom render element. i think we should merge them

File: src/components/molecules/Breadcrumb/BreadcrumbItem.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+                };
+                return cloneElement(renderedElement, { ...propsToApply }, itemContent);
+            }
+        }
+
+        if (isLastItem) {
+            return (
+                <span
+                    className={classNames("breadcrumb__link breadcrumb__link_active", {
+                        breadcrumb__link_iconBefore: !!Icon && !iconOnly,
```
Comment (HaykKirakosyanSoft) [inline src/components/molecules/Breadcrumb/BreadcrumbItem.tsx:84]: this is used in return below too, won't it be more cleaner to make it a variable like let iconbefore = !!Icon && !iconOnly,
